azure-application-insights-spring-boot-starter/README.md

```diff
@@ -0,0 +1,140 @@
+```groovy
+compile "com.microsoft.azure:azure-application-insights-spring-boot-starter:${version}"
+```
+
+Maven:
+```xml
+<dependency>
+  <groupId>com.microsoft.azure</groupId>
+  <artifactId>azure-application-insights-spring-boot-starter</artifactId>
+  <version>${version}</version>
+</dependency>
+```
+
+*2. Provide Instrumentation Key*
+
+Add property
+```
+azure.application-insights.instrumentation-key=<key from the Azure Portal>
+```
+into your `application.properties`
+
+*3. Run your application*
+
+Start your spring boot application as usual and in few minutes you'll start getting events.
+
+**Additional Configuration**
+
+#### Sending custom telemetry
+```java
+@RestController
+public class TelemetryController {
+
+    @Autowired
+    private TelemetryClient telemetryClient;
+
+    @RequestMapping("/telemetry")
+    public void telemetry() {
+        telemetryClient.trackEvent("my event");
+    }
+}
+```
+
+
+#### Sending logs to the application insight
+
+Follow the instructions from [Spring Boot logging documentation](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-logging.html) to configure custom logback or log4j2 appender.
+
+`logback-spring.xml`:
+```xml
+<appender name="aiAppender"
+  class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender">
+</appender>
+<root level="trace">
+  <appender-ref ref="aiAppender" />
+</root>
+```
+
+`log4j2.xml`:
+```xml
+<Configuration packages="com.microsoft.applicationinsights.log4j.v2">
+  <Appenders>
+    <ApplicationInsightsAppender name="aiAppender" />
+  </Appenders>
+  <Loggers>
+    <Root level="trace">
+      <AppenderRef ref="aiAppender"/>
+    </Root>
+  </Loggers>
+</Configuration>
+```
+
+#### Register own telemetry module, processor or initializer by defining it as a bean in the configuration
+```java
+@SpringBootApplication
+public class MyApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(MyApplication.class, args);
+    }
+
+    @Bean
+    public TelemetryModule myTelemetryModule() {
+        return new MyTelemetryModule();
+    }
+
+    @Bean
+    public TelemetryInitializer myTelemetryInitializer() {
+        return new MyTelemetryInitializer();
+    }
+
+    @Bean
+    public TelemetryProcessor myTelemetryProcessor() {
+        return new MyTelemetryProcessor();
+    }
+
+    @Bean
+    public ContextInitializer myContextInitializer() {
+        return new MyContextInitializer();
+    }
+}
+```
+
+
+#### Configure more parameters using `application.properties`
+```properties
+# Enable/Disable tracking
+azure.application-insights.enabled=true
+
+# Instrumentation key from the Azure Portal
+azure.application-insights.instrumentation-key=00000000-0000-0000-0000-000000000000
+
+# Logging type [console, file]
+azure.application-insights.logger.type=console
+# Logging level [all, trace, info, warn, error, off]
+azure.application-insights.logger.level=info
+
+# Enable/Disable default telemetry modules
```

Comment:
By default they are all turned on, added these properties just in case user wants to disable those.

Reply:
Sure, it makes sense to specify all the available properties to tweak in the readme, however please make a note there that all these properties are turned on by default so that user can have an understanding.

Reply:
I was thinking about calling them default already makes them turned on by default ;) I will add note that all those enabled by default